tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1432,6 +1491,67 @@ def _get_dummy_stacker(self, level: int, model_levels: dict, use_orig_features=T
+            'feature_prune_time_limit' to manually specify how long we should perform the feature pruning procedure for, 'k' to specify how long we should
+            perform feature pruning for if 'feature_prune_time_limit' has not been set, and 'raise_exception' to signify that AutoGluon should throw
+            an exception if feature pruning errors out.
+        time_limit : float
+            Time limit left within the current stack layer in seconds. Feature pruning should never take more than this time.
+        layer_fit_time : float
+            How long it took to fit all the models in this layer once. Used to calculate how long to feature prune for.
+        level : int
+            Level of this stack layer.
+        features: List[str]
+            The list of feature names in the inputted dataset.
+
+        Returns
+        -------
+        candidate_features : List[str]
+            Feature names that survived the pruning procedure.
+        """
+        leaderboard = self.leaderboard()
+        leaderboard = leaderboard[~leaderboard['model'].str.contains('WeightedEnsemble')]
+        fit_models = leaderboard[(~leaderboard['score_val'].isna()) & (leaderboard['stack_level'] == level)]
+        if len(fit_models) == 0:
+            return features
+        best_fit_models = fit_models.loc[fit_models['score_val'] == fit_models['score_val'].max()]
+        proxy_model = self.load_model(best_fit_models.loc[best_fit_models['fit_time'].idxmin()]['model'])
+
+        if feature_prune_kwargs[proxy_model.name].pop('proxy_model_class', "LGB") == "LGB":
+            lgb_models = fit_models[fit_models['model'].str.contains('LightGBM')]
+            if len(lgb_models) > 0:
+                best_lgb_models = lgb_models.loc[lgb_models['score_val'] == lgb_models['score_val'].max()]
+                best_lgb_model = best_lgb_models.loc[best_lgb_models['fit_time'].idxmin()]
+                proxy_model = self.load_model(best_lgb_model['model'])
+
+        k = feature_prune_kwargs[proxy_model.name].pop('k', 2)
+        if feature_prune_kwargs[proxy_model.name].get('feature_prune_time_limit', None) is not None:
+            feature_prune_time_limit = min(max(time_limit - layer_fit_time, 0), feature_prune_kwargs[proxy_model.name].pop('feature_prune_time_limit'))
+        elif time_limit is not None:
+            feature_prune_time_limit = min(max(time_limit - layer_fit_time, 0), k * layer_fit_time)
+        else:
+            feature_prune_time_limit = k * layer_fit_time
+        logger.log(30, f"Proxy model feature pruning models for up to {round(feature_prune_time_limit)}s...")
+        raise_exception_on_fail = feature_prune_kwargs[proxy_model.name].pop('raise_exception', False)
+        selector = FeatureSelector(model=proxy_model, time_limit=feature_prune_time_limit, raise_exception=raise_exception_on_fail)
+        candidate_features = selector.select_features(**feature_prune_kwargs[proxy_model.name], **self._model_fit_kwargs[proxy_model.name])
```

Comment:
storing self._model_fit_kwargs is not what we want to be doing. We need a better way to do this before merging.